arch/arm64/core/reboot.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2013-2014 Wind River Systems, Inc.
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ARM Cortex-A and Cortex-R System Control Block interface
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/sys/util.h>
+
+/**
+ *
+ * @brief Reset the system
+ *
+ * This routine resets the processor.
+ *
+ */
+
+void __weak sys_arch_reboot(int type)
+{
+	ARG_UNUSED(type);
+}
```

Comment:
This is not fixing it, it's merely hiding the fact that the platform has no software reboot hook. Applications/samples/tests needing `sys_reboot()` now silently compiles, but cannot expect to run.

I see we have the same weak implementation present in `arch/arm/core/cortex_a_r/reboot.c`, but I don't think that is correct either (CC: @carlocaione, @ithinuel).

Please split this particular change out in it's own PR to get a proper discussion on this.

Reply:
Split as #77056.

Reply:
@CkovMk  as https://github.com/zephyrproject-rtos/zephyr/pull/77905 has been merged, so please decline https://github.com/zephyrproject-rtos/zephyr/pull/77056 and rebase this PR to use PSCI reboot API.